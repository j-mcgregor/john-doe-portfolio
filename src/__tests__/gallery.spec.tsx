import React from 'react'
import { render } from '@testing-library/react'
import { times } from 'lodash'
import * as Gatsby from 'gatsby'
import GalleryPage from '../pages/gallery'
// import { GalleryPageProps } from '../types/interfaces'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')

describe('GalleryPage', () => {
    let props
    beforeEach(() => {
        props = {
            data: {
                allFile: {
                    edges: [
                        {
                            node: {
                                name: 'grid-svg.inline',
                                id: 'grid-svg.inline',
                                publicURL: '/assets/grid-svg.inline.svg',
                            },
                        },
                        {
                            node: {
                                name: 'grid-col-svg.inline',
                                id: 'grid-col-svg.inline',
                                publicURL: '/assets/grid-col-svg.inline.svg',
                            },
                        },
                        {
                            node: {
                                name: 'grid-row-svg.inline',
                                id: 'grid-row-svg.inline',
                                publicURL: '/assets/grid-row-svg.inline.svg',
                            },
                        },
                    ],
                },
                prismic: {
                    allGallerys: {
                        edges: [
                            {
                                node: {
                                    title: [
                                        {
                                            type: 'heading3',
                                            text: 'Gallery',
                                            spans: [],
                                        },
                                    ],
                                    description: [
                                        {
                                            type: 'heading4',
                                            text: 'Look at my stuff!!',
                                            spans: [],
                                        },
                                    ],
                                    body: [
                                        {
                                            type: 'type',
                                            fields: times(10, num => ({
                                                alt_text: [
                                                    {
                                                        type: 'paragraph',
                                                        text: `alt text ${num}`,
                                                        spans: [],
                                                    },
                                                ],
                                                caption: [
                                                    {
                                                        type: 'paragraph',
                                                        text: `caption ${num}`,
                                                        spans: [],
                                                    },
                                                ],
                                                image: {
                                                    dimensions: {
                                                        width: 1000,
                                                        height: 800,
                                                    },
                                                    alt: `alt text ${num}`,
                                                    copyright: `copyright ${num}`,
                                                    url: `http://something-${num}.com`,
                                                },
                                            })),
                                        },
                                    ],
                                },
                            },
                        ],
                    },
                    allContacts: {
                        edges: [
                            {
                                node: {
                                    social_links: [
                                        {
                                            name: 'Mail',
                                            text: 'info@johndoe.com',
                                            url: {
                                                url: 'mailto:info@johndoe.com',
                                            },
                                        },
                                    ],
                                },
                            },
                        ],
                    },
                },
            },
        }
    })

    it('should render with all information', () => {
        useStaticQuery.mockImplementation(() => props.data)
        const { baseElement } = render(<GalleryPage data={props.data} />)

        expect(baseElement).toMatchSnapshot()
    })

    it('should render an empty <div /> if no title', () => {
        const newProps = {
            data: {
                ...props.data,
                prismic: {
                    ...props.data.prismic,
                    allGallerys: {
                        edges: [
                            {
                                node: {
                                    ...props.data.prismic.allGallerys.edges[0]
                                        .node,
                                    title: null,
                                },
                            },
                        ],
                    },
                },
            },
        }
        useStaticQuery.mockImplementation(() => newProps.data)
        const { baseElement } = render(<GalleryPage data={newProps.data} />)
        expect(baseElement).toMatchSnapshot()
    })

    it('should render an empty <div /> if no description', () => {
        const newProps = {
            data: {
                ...props.data,
                prismic: {
                    ...props.data.prismic,
                    allGallerys: {
                        edges: [
                            {
                                node: {
                                    ...props.data.prismic.allGallerys.edges[0]
                                        .node,
                                    description: null,
                                },
                            },
                        ],
                    },
                },
            },
        }
        useStaticQuery.mockImplementation(() => newProps.data)
        const { baseElement } = render(<GalleryPage data={newProps.data} />)
        expect(baseElement).toMatchSnapshot()
    })
})
