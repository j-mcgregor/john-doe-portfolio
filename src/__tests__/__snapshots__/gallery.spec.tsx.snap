// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryPage should render an empty <div /> if no description 1`] = `
<body>
  <div>
    <nav
      class="navbar"
    >
      <a
        class="brand"
        href="/"
      >
        John Doe
      </a>
      <ul
        class="nav-list"
      >
        <li
          class="nav-item"
        >
          <a
            href="/gallery"
          >
            Gallery
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="container-fluid"
    >
      <div
        class="text-center pb3 pt5 px2"
      >
        <h3>
          Gallery
        </h3>
        <div />
      </div>
      <div
        class="gallery-container pb3"
        data-testid="gallery-container"
      >
        <div
          class="container panel"
        >
          <img
            alt="grid-svg.inline"
            data-checked="true"
            data-testid="data-test-svg"
            src="/assets/grid-svg.inline.svg"
            title="data-GRID"
          />
          <img
            alt="grid-col-svg.inline"
            data-checked="false"
            data-testid="data-test-svg"
            src="/assets/grid-col-svg.inline.svg"
            title="data-COL"
          />
          <img
            alt="grid-row-svg.inline"
            data-checked="false"
            data-testid="data-test-svg"
            src="/assets/grid-row-svg.inline.svg"
            title="data-ROW"
          />
        </div>
        <div
          class="modal"
          style="display: none;"
        >
          <span
            class="close"
            data-testid="close"
          >
            ×
          </span>
          <img
            class="modal-content"
            src="http://something-0.com"
          />
          <div
            class="caption"
          />
        </div>
      </div>
    </div>
    <footer
      class="footer"
    >
      Reach me on
      <div
        class="social"
      >
        <a
          href="mailto:info@johndoe.com"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-envelope fa-w-16 "
            data-icon="envelope"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </footer>
  </div>
</body>
`;

exports[`GalleryPage should render an empty <div /> if no title 1`] = `
<body>
  <div>
    <nav
      class="navbar"
    >
      <a
        class="brand"
        href="/"
      >
        John Doe
      </a>
      <ul
        class="nav-list"
      >
        <li
          class="nav-item"
        >
          <a
            href="/gallery"
          >
            Gallery
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="container-fluid"
    >
      <div
        class="text-center pb3 pt5 px2"
      >
        <div />
        <h4>
          Look at my stuff!!
        </h4>
      </div>
      <div
        class="gallery-container pb3"
        data-testid="gallery-container"
      >
        <div
          class="container panel"
        >
          <img
            alt="grid-svg.inline"
            data-checked="true"
            data-testid="data-test-svg"
            src="/assets/grid-svg.inline.svg"
            title="data-GRID"
          />
          <img
            alt="grid-col-svg.inline"
            data-checked="false"
            data-testid="data-test-svg"
            src="/assets/grid-col-svg.inline.svg"
            title="data-COL"
          />
          <img
            alt="grid-row-svg.inline"
            data-checked="false"
            data-testid="data-test-svg"
            src="/assets/grid-row-svg.inline.svg"
            title="data-ROW"
          />
        </div>
        <div
          class="modal"
          style="display: none;"
        >
          <span
            class="close"
            data-testid="close"
          >
            ×
          </span>
          <img
            class="modal-content"
            src="http://something-0.com"
          />
          <div
            class="caption"
          />
        </div>
      </div>
    </div>
    <footer
      class="footer"
    >
      Reach me on
      <div
        class="social"
      >
        <a
          href="mailto:info@johndoe.com"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-envelope fa-w-16 "
            data-icon="envelope"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </footer>
  </div>
</body>
`;

exports[`GalleryPage should render with all information 1`] = `
<body>
  <div>
    <nav
      class="navbar"
    >
      <a
        class="brand"
        href="/"
      >
        John Doe
      </a>
      <ul
        class="nav-list"
      >
        <li
          class="nav-item"
        >
          <a
            href="/gallery"
          >
            Gallery
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="container-fluid"
    >
      <div
        class="text-center pb3 pt5 px2"
      >
        <h3>
          Gallery
        </h3>
        <h4>
          Look at my stuff!!
        </h4>
      </div>
      <div
        class="gallery-container pb3"
        data-testid="gallery-container"
      >
        <div
          class="container panel"
        >
          <img
            alt="grid-svg.inline"
            data-checked="true"
            data-testid="data-test-svg"
            src="/assets/grid-svg.inline.svg"
            title="data-GRID"
          />
          <img
            alt="grid-col-svg.inline"
            data-checked="false"
            data-testid="data-test-svg"
            src="/assets/grid-col-svg.inline.svg"
            title="data-COL"
          />
          <img
            alt="grid-row-svg.inline"
            data-checked="false"
            data-testid="data-test-svg"
            src="/assets/grid-row-svg.inline.svg"
            title="data-ROW"
          />
        </div>
        <div
          class="modal"
          style="display: none;"
        >
          <span
            class="close"
            data-testid="close"
          >
            ×
          </span>
          <img
            class="modal-content"
            src="http://something-0.com"
          />
          <div
            class="caption"
          />
        </div>
      </div>
    </div>
    <footer
      class="footer"
    >
      Reach me on
      <div
        class="social"
      >
        <a
          href="mailto:info@johndoe.com"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-envelope fa-w-16 "
            data-icon="envelope"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </footer>
  </div>
</body>
`;
